boa_engine/src/vm/mod.rs

```diff
@@ -279,14 +252,12 @@ impl Context<'_> {
             stack
         };
 
-        println!(
-            "{:<TIME_COLUMN_WIDTH$} {:<OPCODE_COLUMN_WIDTH$} {operands:<OPERAND_COLUMN_WIDTH$} {stack}",
-            format!("{}μs", duration.as_micros()),
-            opcode.as_str(),
-            TIME_COLUMN_WIDTH = Self::TIME_COLUMN_WIDTH,
-            OPCODE_COLUMN_WIDTH = Self::OPCODE_COLUMN_WIDTH,
-            OPERAND_COLUMN_WIDTH = Self::OPERAND_COLUMN_WIDTH,
-        );
+        if let Some(trace) = &self.vm.trace {
+            #[cfg(not(target_arch = "wasm32"))]
+            trace.trace_instruction(duration.as_micros(), opcode.as_str(), &operands, &stack);
+            #[cfg(target_arch = "wasm32")]
+            trace.trace_instruction(opcode.as_str(), &operands, &stack);
+        }
```

Comment:
For simplicity, I think we should just have a `Trace` type that doesn't print anything. That would allow us to remove the `Option` handling and just directly call `trace_instruction`.